python_modules/dagster/dagster/_core/execution/plan/objects.py

```diff
@@ -79,18 +83,37 @@ class ErrorSource(Enum):
     INTERRUPT = "INTERRUPT"
 
 
-@whitelist_for_serdes
+class AssetKeyCompactSerializer(NamedTupleSerializer):
```

Comment:
Maybe this is premature optimization, but I was imagining that the serdes representation for an AssetKey NamedTuple is quite a bit less compact than the string representation, and a p99 step could have thousands of outputs

Reply:
would be good to pair this with a test that measures the size of the serialized step event to demonstrate how useful this is when there are ~5000 assets